packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java

```
@@ -213,25 +225,66 @@ static Integer convertMediaType(String mediaType) throws IllegalArgumentExceptio
     return MEDIA_TYPES.get(mediaType);
   }
 
-  static int[] getSizeFromCursor(Cursor cursor, int mediaType, int localUriIndex){
+  static int[] getSizeFromCursor(ContentResolver contentResolver, ExifInterface exifInterface, Cursor cursor, int mediaType, int localUriIndex) {
+    final String uri = cursor.getString(localUriIndex);
+
+    if (mediaType == Files.FileColumns.MEDIA_TYPE_VIDEO) {
+      Uri photoUri = Uri.parse("file://" + uri);
+      try {
+        AssetFileDescriptor photoDescriptor = contentResolver.openAssetFileDescriptor(photoUri, "r");
+        MediaMetadataRetriever retriever = new MediaMetadataRetriever();
+        try {
+          retriever.setDataSource(photoDescriptor.getFileDescriptor());
+          int videoWidth = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_WIDTH)
+          );
+          int videoHeight = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_HEIGHT)
+          );
+          int videoOrientation = Integer.parseInt(
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION)
+          );
+          return maybeRotateAssetSize(videoWidth, videoHeight, videoOrientation);
+        } catch (NumberFormatException e) {
```

Maybe instead of ignoring this exception, we should log it.